virttest/env_process.py

```diff
@@ -509,7 +512,35 @@ def process(test, params, env, image_func, vm_func, vm_first=False):
     :param vm_func: A function to call for each VM.
     :param vm_first: Call vm_func first or not.
     """
+    def validate_memory_resource():
+        """Validate host has enough memory to lauch VMs"""
+        magnification, requried_mem, count = 1.0, 0, 1
+        for vm_name in params.objects("vms"):
+            vm_params = params.object_params(vm_name)
+            if vm_params.get("start_vm") == "yes":
+                mem = "%sM" % vm_params.get("mem", 512)
+                requried_mem += float(
+                    utils_misc.normalize_data_size(mem))
+                count += 1
+        if (params.get("setup_ksm") == "yes" and
+                params.get("ksm_run", "1") == "1"):
+            magnification = 1.2
+        free_mem = "%s KB" % memory.freememtotal()
+        free_mem = float(utils_misc.normalize_data_size(free_mem))
+        provide_mem = free_mem * magnification
+        provide_mem = float(utils_misc.normalize_data_size(provide_mem))
```

Comment:
00:55:36 ERROR|   File "/usr/lib/python2.6/site-packages/avocado_plugins_vt-44.0-py2.6.egg/virttest/env_process.py", line 531, in validate_memory_resource
00:55:36 ERROR|     provide_mem = float(utils_misc.normalize_data_size(provide_mem))
00:55:36 ERROR|   File "/usr/lib/python2.6/site-packages/avocado_plugins_vt-44.0-py2.6.egg/virttest/utils_misc.py", line 2582, in normalize_data_size
00:55:36 ERROR|     match = re.search(regex, value_str)
00:55:36 ERROR|   File "/usr/lib64/python2.6/re.py", line 142, in search
00:55:36 ERROR|     return _compile(pattern, flags).search(string)
00:55:36 ERROR| TypeError: expected string or buffer